crmsh/qdevice.py

```diff
@@ -173,20 +245,24 @@ def valid_qnetd(self):
         """
         Validate on qnetd node
         """
-        if utils.check_ssh_passwd_need(self.qnetd_addr):
-            self.askpass = True
-
         exception_msg = ""
         suggest = ""
         if utils.service_is_active("pacemaker", self.qnetd_addr):
-            exception_msg = "host for qnetd must be a non-cluster node"
+            exception_msg = "Host for qnetd must be a non-cluster node!"
             suggest = "change to another host or stop cluster service on {}".format(self.qnetd_addr)
         elif not utils.package_is_installed("corosync-qnetd", self.qnetd_addr):
-            exception_msg = "Package \"corosync-qnetd\" not installed on {}".format(self.qnetd_addr)
+            exception_msg = "Package \"corosync-qnetd\" not installed on {}!".format(self.qnetd_addr)
             suggest = "install \"corosync-qnetd\" on {}".format(self.qnetd_addr)
-
+        if utils.service_is_active("corosync-qnetd", self.qnetd_addr):
+            cmd = "corosync-qnetd-tool -l -c {}".format(self.cluster_name)
```

Comment:
This adds assumption qnetd server must be running. This might be a little bit too strong requirement?!

One use case comes to my mind. It is to populate certifications without a running qnetd server. This is sounds a valid. To do that, keeping cluster-$ClusterName.crt on qnetd server seems a reasonable approach to replace dynamic detection above.

Reply:
Here the dynamic checking by corosync-qnetd-tool  command, I think it's enough for more generic situation, which all related nodes and servers are running during setup process

I already removed the crt file after finished certification process since I think it's might not safe

And even we keep that file to do the static checking, there always has corner case

Reply:
BTW, instead of consider all the corner cases, I suggest we make a check when running crm cluster start
That means, when need to start qdevice, after the starting action finished, we check:
If qdevice process started
If qdevice give the vote(by checking corosync-quorumtool) , I found that when have duplicated cluster name, later cluster cannot get the qdevice vote
If that vote in qdevice is 0 or qdevice process not started but qdevice configured, crm cluster start  give warning to user